lib/components/Input/TextInputs.jsx

```diff
@@ -0,0 +1,122 @@
+import React from 'react'
+import classnames from 'classnames'
+import { omit } from 'lodash'
+import { isBrowser } from 'react-device-detect'
+
+import { DEFAULT_INPUT_CLASSES } from '../../constants'
+
+const sanitizeProps = (props) => {
+  return omit(props, [
+    'alignLeft',
+    'label',
+    'small',
+    'large',
+    'marginClasses',
+    'paddingClasses',
+    'borderClasses',
+    'bgClasses',
+    'inlineButton',
+    'roundedClasses',
+    'textClasses',
+    'isError',
+    'isLight',
+    'register',
+    'required', // required is consumed by the register func but we don't want it on the <input />
+    'pattern',
+    'validate',
+    'unsignedNumber',
+    'unsignedWholeNumber',
+    'rightLabel',
+    'bottomRightLabel'
+  ])
+}
+
+const collectClassNames = (props) => {
+  return classnames(
+    DEFAULT_INPUT_CLASSES,
```

Comment:
I kinda dislike this pattern for styling. If we want consistent styles we need to copy several props and update all over if we want to change styles.

It'd be much simpler if we just had a `theme` prop where we passed `theme={TextInputsTheme.primary}` which determines what all of these different sets of class names are.

```
const TextInputsTheme = Object.freeze({
   primary: 'primary',
   secondary: 'secondary',
   inverse: 'inverse',
})

const TextInputsSize = Object.freeze({
   small 'small',
   default: 'default'
   large: 'large',
})
```

Not urgent at all, just a comment for future improvements.

Reply:
💯 I agree, the current way we do styling is pretty disastrous. That said, I think that falls into refactoring. We should spend time on this soon, perhaps after we launch the Tsunami prototype.